apps/spectator/queries/useGameList.ts

```diff
@@ -0,0 +1,67 @@
+import { useSuspenseInfiniteQuery } from '@tanstack/react-query';
+
+import { getGameList } from '@/api/game';
+import { GameListParams, GameListType } from '@/types/game';
+
+export const useGameList = ({
+  sport_id,
+  state,
+  league_id,
+  league_team_id,
+  round,
+}: Omit<GameListParams, 'cursor' | 'size'>) => {
+  const { data, error, fetchNextPage, hasNextPage, isFetching } =
+    useSuspenseInfiniteQuery({
+      queryKey: [
+        'game-list',
+        sport_id,
+        state,
+        league_id,
+        league_team_id,
+        round,
+      ],
+      initialPageParam: 0,
+      queryFn: ({ pageParam }) =>
+        getGameList({
+          sport_id,
+          state,
+          league_id,
+          league_team_id,
+          round,
+          cursor: pageParam || '',
+        }),
+      getNextPageParam: lastPage => lastPage[lastPage.length - 1]?.id || null,
+      select: data => {
+        const gameMap = new Map<string, GameListType[]>();
+
+        data.pages.flatMap(gameList => {
+          gameList.forEach(game => {
+            const date = game.startTime.split('T')[0];
+            const existing = gameMap.get(date);
+
+            if (existing) {
+              existing.push(game);
+              gameMap.set(date, existing);
```

Comment:
```suggestion
gameMap.set(date, [...existing, ...game])
```

불변성을 지키기 위해 위와 같은 방법은 어떨까요?

Reply:
game의 경우 GameListType으로 순회가능한 타입은 아니다보니 해당 변수만 spread 연산자 미적용해서 반영하겠습니다~!

Reply:
아하 그러네요ㅎㅎ 좋습니다👍👍